bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -255,20 +262,23 @@ where
 		match dest_para_id {
 			Some(dest_para_id) => {
 				let dest_para_fee_asset: Asset = (Location::parent(), dest_para_fee).into();
+				let bridge_location = Location::new(2, GlobalConsensus(network));
 
 				instructions.extend(vec![
 					// Perform a deposit reserve to send to destination chain.
 					DepositReserveAsset {
-						assets: Definite(vec![dest_para_fee_asset.clone(), asset.clone()].into()),
+						assets: Definite(vec![dest_para_fee_asset.clone(), asset].into()),
 						dest: Location::new(1, [Parachain(dest_para_id)]),
 						xcm: vec![
 							// Buy execution on target.
 							BuyExecution { fees: dest_para_fee_asset, weight_limit: Unlimited },
-							// Deposit asset to beneficiary.
-							DepositAsset { assets: Definite(asset.into()), beneficiary },
+							// Deposit assets to beneficiary.
+							DepositAsset { assets: Wild(AllCounted(2)), beneficiary },
 						]
 						.into(),
 					},
+					// Deposit any leftover unspent AH fees to the snowbridge sovereign.
+					DepositAsset { assets: Wild(AllCounted(1)), beneficiary: bridge_location },
```

Comment:
I would suggest to move this before `DepositReserveAsset` with a `SetAppendix`

Reply:
let's see difference in effects:

On XCM success, with or without `SetAppendix` nothing differs (slightly more expensive with `SetAppendix`, but not important): all ERC20 sent to dest, DOT spent for fees, surplus DOT deposited to snowbridge SA

On XCM error (`DepositReserveAsset` error):
- a) without `SetAppendix`: all ERC20 and all DOT get trapped
- b) with `SetAppendix(DepositAsset(Wild(AllCounted(1))))`: all ERC20 get trapped, all DOT deposited to snowbridge SA
- c) with `SetAppendix(DepositAsset(Wild(AllCounted(2))))`: all ERC20 and all DOT deposited to snowbridge SA

Which one do you prefer? I would go with either a) or c) for consistency. I have slight preference for c) because it's slightly easier to recover, no need to claim. But in both cases, governance is required to recover the funds.

Reply:
I prefer c and that's also what I do in PNA.
https://github.com/Snowfork/polkadot-sdk/blob/34c655ba7bf8192b417f76e19be3d7d2435a3fcf/bridges/snowbridge/primitives/router/src/inbound/mod.rs#L443-L447